frame/contracts/src/wasm/runtime.rs

```diff
@@ -1138,8 +1155,16 @@ define_env!(Env, <E: Ext>,
 	// the caller contract and restore the destination contract and set the specified `rent_allowance`.
 	// All caller's funds are transfered to the destination.
 	//
+	// The tombstone hash is derived as `hash(code_hash, storage_root_hash)`. In order to match
+	// this hash to its own hash the restorer must make its storage equal to the one of the
+	// evicted destination contract. In order to allow for additional storage items in the
+	// restoring contract a delta can be specified to this function. All keys specified as
+	// delta are disregarded when calculating the storage root hash.
+	//
 	// If there is no tombstone at the destination address, the hashes don't match or this contract
-	// instance is already present on the contract call stack, a trap is generated.
+	// instance is already present on the contract call stack, a trap is generated. Additionally,
```

Comment:
I think this would be more readable as a list of trap conditions e.g.

This function will trap if:

- there is no tombstone at the destination address
- the hashes don't match
- this contract instance is already present on the contract call stack
- the code hash was also (already?) evicted

Reply:
I already have this list. I removed the whole paragraph and improved the list of traps. This paragraph was confusing.